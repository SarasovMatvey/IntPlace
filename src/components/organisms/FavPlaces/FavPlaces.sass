.fav-places
  &__card
    position: relative
  &__card-delete-btn
    position: absolute
    left: 0
    top: 0
    transform: translate(-50%, -50%)
    z-index: 6
    background: #ff9800 !important
    color: #fff !important
    display: none !important
  &__card:hover &__card-delete-btn
    display: block !important