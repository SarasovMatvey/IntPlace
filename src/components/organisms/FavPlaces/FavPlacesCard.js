import { Button } from "semantic-ui-react";
import PlaceCard from "../../atoms/PlaceCard";

function FavPlacesCard({
  description,
  onClick,
  placeId,
  poster,
  title,
  onDeleteBtnClick,
}) {
  return (
    <div className="fav-places__card">
      <PlaceCard
        className="fav-places__card-inner"
        fluid
        description={description}
        onClick={onClick}
        placeId={placeId}
        poster={poster}
        title={title}
      >
        <Button
          className="fav-places__card-delete-btn"
          circular
          icon="close"
          onClick={onDeleteBtnClick}
        />
      </PlaceCard>
    </div>
  );
}

export default FavPlacesCard;
