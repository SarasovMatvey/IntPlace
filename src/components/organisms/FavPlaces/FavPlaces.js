import Masonry from "react-responsive-masonry";
import FavPlacesCard from "./FavPlacesCard";
import sygicAxios from "../../../api/sygic";
import { useState } from "react";
import { useEffect } from "react";
import { useHistory } from "react-router";
import "./FavPlaces.sass";

function FavPlaces() {
  const history = useHistory();
  const [places, setPlaces] = useState([]);

  useEffect(() => {
    const favPlacesIds = JSON.parse(localStorage.getItem("favPlaces"));
    sygicAxios(
      `/places?ids=${favPlacesIds.join("%7C")}`
    ).then(({ data: { data } }) => setPlaces(data.places));
  }, []);

  return (
    <div className="fav-places">
      <Masonry gutter="2rem" columnsCount={4}>
        {places.map(placeInfo => (
          <FavPlacesCard
            key={placeInfo.id}
            title={placeInfo.name}
            poster={placeInfo.thumbnail_url}
            description={placeInfo.perex}
            placeId={placeInfo.id}
            onClick={e => goToPlaceInfoPage(e.currentTarget.dataset.id)}
            onDeleteBtnClick={e => {
              e.stopPropagation();
              deleteFavPlace(e.currentTarget.dataset.id);
            }}
          />
        ))}
      </Masonry>
    </div>
  );

  function deleteFavPlace(placeId) {
    const favPlaces = JSON.parse(localStorage.getItem("favPlaces"));
    const deletedPlaceIndex = favPlaces.indexOf(placeId);
    favPlaces.splice(deletedPlaceIndex, 1);
    setPlaces(places => {
      const newPlaces = places.splice(deletedPlaceIndex, 1);
      return newPlaces;
    });

    localStorage.setItem("favPlaces", JSON.stringify(favPlaces));
  }

  function goToPlaceInfoPage(placeId) {
    history.push(`place/${placeId}`);
  }
}

export default FavPlaces;
