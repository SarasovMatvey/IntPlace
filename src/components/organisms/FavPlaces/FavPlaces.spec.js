import FavPlaces from "./FavPlaces";

const setUp = props => shallow(<FavPlaces {...props} />);

describe("FavPlaces", () => {
  let component;

  beforeEach(() => {
    component = setUp();
  });

  it("should renders properly", () => {
    expect(component).toMatchSnapshot();
  });
});
