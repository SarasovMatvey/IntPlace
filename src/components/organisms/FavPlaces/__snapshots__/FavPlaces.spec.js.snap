// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavPlaces should renders properly 1`] = `
<div
  className="fav-places"
>
  <Masonry
    className={null}
    columnsCount={4}
    gutter="2rem"
    style={Object {}}
  />
</div>
`;
